### PR TITLE
Update DG with state feature, refactor state enum, fix lesson card

### LINKED ISSUE #288: Keywords in list method affects schedule list
Typing `list students subjects` or `list students all` will have subjects showing in each lesson in schedule list

### LINKED ISSUE #321: refactor state evaluation in command result

### PR DESCRIPTION
**Pull request by pzl111**
Update DG with state feature
Add AddSequenceDiagram puml
Remove "task" command word for addTask
Modify AddressBookParser to use model.sameState()
Edit test case for addTask
Update LinkCommand to use sameState() method

Fixes #321 
Fixes #288  (Lesson card having displayedFields)

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddTaskCommand.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ private void checkClash(Model model, int lessonIndex) throws CommandException {
         }
     }
     public static String getUsageInfo() {
-        return "\nUsage: addTask/task + [lesson index] [description]. "
```
Comment (yiwen101) [inline src/main/java/seedu/address/logic/commands/AddTaskCommand.java:99]: Please also be reminded to change the usage message in the failure output for addTask in TempUG

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 2]
```diff
@@ -135,22 +135,16 @@ public Command parseCommand(String userInput) throws ParseException {
...
-            }
-            return new AddTaskCommandParser().parse(arguments);
-        case "task":
-            if (!model.getState().equals(SCHEDULE)) {
+            if (!model.sameState(SCHEDULE)) {
                 throw new ParseException("Please add tasks in the schedule list.");
             }
             return new AddTaskCommandParser().parse(arguments);
         case DeleteTaskCommand.COMMAND_WORD:
-            if (!model.getState().equals(SCHEDULE)) {
```
Comment (yiwen101) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:148]: emm, could you use the refractor tool to globally rename this method to "isSameState"? 
Not very important though

**Review by yiwen101 — CHANGES_REQUESTED** [review]
LGTM
Very impressive effort and significant code quality improvement

File: src/main/java/seedu/address/logic/commands/AddTaskCommand.java (hunk 1 above)
Comment (pzl111) [reply src/main/java/seedu/address/logic/commands/AddTaskCommand.java:99]: okay corrected it

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java (hunk 2 above)
Comment (pzl111) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java:148]: okay corrected it

**Review by pzl111 — COMMENTED** [review]
okay made changes

**Review by yiwen101 — APPROVED** [review]

**Pull request merged**